### ISSUE TITLE
SPI does not work in Slovenia

### ISSUE THREAD

**Issue by andimik** [softsyst/qirx_issues#143]
Mentioned in https://www.rundfunkforum.de/viewtopic.php?p=1747230#p1747230

Although two muxes in Slovenia (R2W and R2E) have introduced a real EPG within SPI, no files are stored in the SPI folder.

![Screenshot (433)](https://github.com/softsyst/qirx_issues/assets/24510556/23ad1bfe-045a-4d29-a529-11901ac89843)

AbracaDABra has no problems with the data.

![Bildschirmfoto vom 2024-01-31 21-33-58](https://github.com/softsyst/qirx_issues/assets/24510556/71825381-ef63-4200-bcb0-7aceb5c92956)

Maybe the following log helps?

[qirx.logqirx.log-2024-02-01--00-13-31.txt.zip](https://github.com/softsyst/qirx_issues/files/14226212/qirx.logqirx.log-2024-02-01--00-13-31.txt.zip)

Remark: As you can see, none of the services got a station logo

**Comment by softsyst** [comment]
Yes, the log helps. I had a look at this file, and it shows tons of MOT packet errors, crc Errors etc. QIRX will never be able to decode such a stream. 

An example from the log, as displayed by log4view:

![image](https://github.com/softsyst/qirx_issues/assets/51386375/547f1927-1b3e-43cb-b794-3384e754946b)

I am not familiar with AbracaDABra. If you want to have the behavior in QIRX investigated, I need some raw file (a raw, not an ETI) of sufficient length. Otherwise, no chance.

**Comment by andimik** [comment]
Qirx has no problems with the audio, the signal is 100%, see screenshot.

**Comment by softsyst** [comment]
See the last remark of my last posting.

**Comment by andimik** [comment]
It takes around 10 hours to send you the 2 GB raw file via wetransfer. Please be patient.

**Comment by andimik** [comment]
I really encourage you to install ODR-DabMod, as the 2 GB raw-file is just 58 MB in an eti (compressed).

https://get.hidrive.com/i/Zk6XJ8ZJ
(58 MB eti, zipped)

In case the 150 MB raw is enough, it's here.

https://get.hidrive.com/i/VSxF9eyP

If not, I need to send you the 2 GB on a DVD probably per Post. I tried several times on various file hosters, I had no luck and it always ended in an error before EOF.

**Comment by softsyst** [comment]
Thanks for the files. I try with the 150MB one. If it is too small, it will take some time, as I have currently no time to tinker around with the ODR tools.

To be honest, I have a problem understanding your reasoning, as the (popular!) Google Dropbox allows 2G. I used it very often for files above 1G. Are the 2G necessary? Does it really take almost 10 minutes to complete the EPG in your mux?

**Comment by andimik** [comment]
Yes of course, otherwise I would have not mentioned 2 GB.

**Comment by softsyst** [comment]
I sent you a PM, with an alternative suggestion for Upload of your file.

**Comment by softsyst** [comment]
@andimik ,

Thanks for uploading a raw file of about 10 minutes, Multiplex E4-9400 SLO DAB+ R2 W. 

I analyzed your file, with the following result:
- MOT packets arrive, no sequence or CRC errors were reported during the whole transmission of about 10 minutes.
- All MOT objects are of the data group Type "MOT Body". For a decoding by QIRX this is not sufficient. QIRX decodes only SPI objects in the "Directory Mode". This requires also objects of the data group type "MOT Directory".
- No object of the (necessary) data group type "MOT Directory" was observed during the file reception.
- The SPI Standard ETSI TS 102 371 V3.2.1 (2016-05) requires in 6.1 "Transport mechanism":
"MOT in Directory Mode (ETSI EN 301 234 [2]) will be employed as the method for transporting the SPI data."

Also the relaxed "MOT Header Mode" for general MOT Objects requires a corresponding data Group type of "MOT Header"., which was not observed during the reception (and is not handled by QIRX).

**To summarize, as QIRX works for SPI decoding only in Directory Mode, as required by the Standard, the SPI content in the file will not be decoded by QIRX.**

**Remark**: The "Blinking Lamps" on the GUI, usually indicating an ongoing collection of SPI objects, cannot be omitted, because an object of data group type "MOT Directory" can arrive at any point in time, unknown by the software.

**Comment by softsyst** [comment]
No new comments, I close it.

**Comment by andimik** [comment]
Well, my comment is, I have technically no idea here and need to trust you.

**Comment by andimik** [comment]
Good news, the broadcaster (RTV Slovenija) has fixed the issue. For a while there was no SPI transmitted at all, but now it works also in Qirx (at least for the Logos, the EPG does not work in Qirx, although there is a preview, which I can see in AbracaDABra).

![grafik](https://github.com/user-attachments/assets/e61d5931-fae9-48ef-80b9-0da951575e07)
![grafik](https://github.com/user-attachments/assets/73af4506-c838-4a73-b3f9-fb6eb4e04063)

The header in the old xml file was

`<serviceInformation creationTime="2024-07-08 09:27:56" originator="AVT MAGIC DABMUX">`

now it is

`<serviceInformation creationTime="2024-10-28 08:32:11">`

Probably they had a hardware upgrade recently, which finally seems to be compliant with the standard.
